aten/src/THC/generic/THCTensorMathMagma.cu

```diff
@@ -12,7 +12,9 @@ static void THCTensor_(copyArray1d)(THCState *state, THCTensor *self, scalar_t *
   int64_t stride[1] = { 1 };
   THCTensor_(resizeNd)(state, self, 1, size, stride);
   size_t len = k * sizeof(scalar_t);
-  THCudaCheck(cudaMemcpy(THCStorage_(data)(state, THTensor_getStoragePtr(self)) + self->storage_offset(), src, len, cudaMemcpyHostToDevice));
+  auto stream = c10::cuda::getCurrentCUDAStream();
+  THCudaCheck(cudaMemcpyAsync(THCStorage_(data)(state, THTensor_getStoragePtr(self)) + self->storage_offset(), src, len, cudaMemcpyHostToDevice, stream));
+  AT_CUDA_CHECK(cudaStreamSynchronize(stream));
```

Comment:
This is functionally correct, and in most cases magma performance is pretty bad anyway so probably we should not micro-optimize, but if pytorch is currently on default stream you don't need a synchronization here, right?

Reply:
This needs to complete before the host memory is freed immediately afterwords.

Reply:
See it's use on line 296.